src/containers/WorkspaceView.tsx

```diff
@@ -1,50 +1,184 @@
-        <Splitter
-            {...{width, height}}
-            initialPosition={300}
-            positionFromRight={true}
-        >
-            <div>
-                {
-                    !showImportInstructions && (
-                        <div><Button onClick={handleImport}>Import recordings</Button></div>
-                    )
-                }
-                <RecordingsTable
-                    {...{sortings, recordings, onDeleteRecordings, onSetRecordingInfo, workspaceInfo}}
-                />
-            </div>
-            {
-                showImportInstructions && (
-                    <ImportRecordingsInstructions />
-                )
-            }
-        </Splitter>
+type Page = 'recordings' | 'recording' | 'sorting'
+const isPage = (x: string): x is Page => {
+    return ['recordings', 'recording', 'sorting'].includes(x)
+}
+
+type WorkspaceRecordingsRoute = {
+    page: 'recordings',
+    workspaceName: string
+}
+type WorspaceRecordingRoute = {
+    page: 'recording',
+    recordingId: string,
+    workspaceName: string
+}
+type WorspaceSortingRoute = {
+    page: 'sorting',
+    recordingId: string,
+    sortingId: string,
+    workspaceName: string
+}
+type WorkspaceRoute = WorkspaceRecordingsRoute | WorspaceRecordingRoute | WorspaceSortingRoute
+type GotoRecordingsPageAction = {
+    type: 'gotoRecordingsPage'
+}
+type GotoRecordingPageAction = {
+    type: 'gotoRecordingPage',
+    recordingId: string
+}
+type GotoSortingPageAction = {
+    type: 'gotoSortingPage',
+    recordingId: string,
+    sortingId: string
+}
+type WorkspaceRouteAction = GotoRecordingsPageAction | GotoRecordingPageAction | GotoSortingPageAction
+export type WorkspaceRouteDispatch = (a: WorkspaceRouteAction) => void
+
+const routeFromLocation = (location: {pathname: string, search: string}): WorkspaceRoute => {
+    const pathList = location.pathname.split('/')
+    const workspaceName = pathList[1] || 'default'
+    const page = pathList[2] || 'recordings'
+    if (!isPage(page)) throw Error(`Invalid page: ${page}`)
+    switch (page) {
+        case 'recordings': return {
+            workspaceName,
+            page
+        }
+        case 'recording': return {
+            workspaceName,
+            page,
+            recordingId: pathList[3]
+        }
+        case 'sorting': return {
+            workspaceName,
+            page,
+            recordingId: pathList[3] || '',
+            sortingId: pathList[4] || ''
+        }
+        default: return {
+            workspaceName,
+            page: 'recordings'
+        }
+    }
+}
+
+var queryString = (params: {[key: string]: string}) => {
+    const keys = Object.keys(params)
+    if (keys.length === 0) return ''
+    return '?' + (
+        keys.map((key) => {
+            return encodeURIComponent(key) + '=' + encodeURIComponent(params[key])
+        }).join('&')
     )
 }
 
+const locationFromRoute = (route: WorkspaceRoute, workspaceInfo: WorkspaceInfo) => {
```

Comment:
Convert a route (as defined above) to a Location object, which is essentially the page URL